bndtools.core/test/bndtools/wizards/repo/RepoBundleVersionToRangeTest.java

```diff
@@ -0,0 +1,93 @@
+package bndtools.wizards.repo;
+
+import static aQute.bnd.version.VersionRange.parseVersionRange;
+import static bndtools.model.repo.DependencyPhase.Build;
+import static bndtools.model.repo.DependencyPhase.Req;
+import static bndtools.model.repo.DependencyPhase.Run;
+import static bndtools.model.repo.RepositoryBundleUtils.convertRepoBundle;
+import static bndtools.model.repo.RepositoryBundleUtils.convertRepoBundleVersion;
+import static bndtools.model.repo.RepositoryBundleUtils.toVersionRangeUpToNextMajor;
+import static bndtools.model.repo.RepositoryBundleUtils.toVersionRangeUpToNextMicro;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+
+import org.junit.jupiter.api.Test;
+
+import aQute.bnd.build.WorkspaceRepository;
+import aQute.bnd.service.RepositoryPlugin;
+import aQute.bnd.version.Version;
+import bndtools.model.repo.RepositoryBundle;
+import bndtools.model.repo.RepositoryBundleVersion;
+
+/**
+ * Tests behavior involved when dragging / dropping / adding repobundles to
+ * -runbundles, -runrequires in the UI e.g. .bndrun editor. In these cases a
+ * simple bsn or bsn+version needs to be converted to a range (or not). This
+ * test tests the different combinations. e.g.
+ * <ul>
+ * <li>foo.bundle.bar v1.2.3 -> foo.bundle.bar;version=[1.2.3,1.2.4] (micro
+ * version increment)</li>
+ * <li>foo.bundle.bar v1.2.3 -> foo.bundle.bar;version=[1.2.3,2.0.0] (major
+ * version increment)</li>
+ * </ul>
+ */
```

Comment:
I highly recommend using AssertJ and the SoftAssertionsExtension for this, especially as your tests have multiple assertions each.

Reply:
Done. Thanks for the tip. Learned something new 👍